src/Controller/SubmissionController.php

```diff
@@ -255,4 +255,55 @@ public function lock(
             'slug' => Slugger::slugify($submission->getTitle()),
         ]);
     }
+
+    /**
+     * @IsGranted("ROLE_USER")
+     *
+     * @param EntityManager $em
+     * @param Request       $request
+     * @param Forum         $forum
+     * @param Submission    $submission
+     *
+     * @return Response
+     */
+    public function report(
+        EntityManager $em,
+        Request $request,
+        Forum $forum,
+        Submission $submission
+    ) {
+        $this->validateCsrf('report_submission', $request->request->get('token'));
+
+        // Find nautbot.
+        $nautbot = $em->find("App\\Entity\\User", 0);
+
+        if($nautbot != null) {
+            $reportSubmission = new Submission(
+                "Submission Report: " . $submission->getTitle(),
+                "/f/" . $forum->getName() . "/" . $submission->getId() . "/" . Slugger::slugify($submission->getTitle()),
+                null,
+                $forum,
+                $nautbot,
+                $request->getClientIp()
+            );
+            $reportSubmission->setModThread(true);
+
+            $em->persist($reportSubmission);
+            $em->flush();
+
+            $this->addFlash('success', 'flash.thread_reported');
+        } else {
+            $this->addFlash('notice', 'flash.generic_error');
+        }
+
+        if ($request->headers->has('Referer')) {
+            return $this->redirect($request->headers->get('Referer'));
+        }
+
+        return $this->redirectToRoute('submission', [
+            'forum_name' => $forum->getName(),
+            'submission_id' => $submission->getId(),
+            'slug' => Slugger::slugify($submission->getTitle()),
+        ]);
+    }
```

Comment:
I'm working on a new Utils class for NautBot functions
src/Utils/NautBot.php
It'd be sweet if you used the same and build something like
```
newNautBotNotice($forum, $title, $body): Submission {
    // build nautbot submission and return
}
```
That returned a new submission. This would get used a lot and be a huge help!

Reply:
Not seeing that class. Is it in another branch?

Reply:
It's not pushed up yet, still working on this feature. Sorry I should've been more clear.